### ISSUE TITLE
"Run-Time Check Failure #0" error when starting

### ISSUE THREAD

**Issue by Megavoxel01** [Celtoys/Remotery#117]
Hi!

I got the following error when I'm trying to profiling OpenGL program.
"Run-Time Check Failure #0 - The value of ESP was not properly saved across a function call.  This is usually a result of calling a function declared with one calling convention with a function pointer declared with a different calling convention."
Program crashed at line 
opengl->__glGenQueries = (PFNGLGENQUERIESPROC)rmtglGetProcAddress(opengl, "glGenQueries") ;
in the _rmt_BindOpenGL() function 
I'm compiling on VS 2015, Windos 10.  Is there anything I may do wrong? Thanks for any help in advance.

**Comment by dwilliamson** [comment]
I've not seen that one, no. I only use the OpenGL myself on OSX.

There's [line 6035](https://github.com/Celtoys/Remotery/blob/master/lib/Remotery.c#L6035) which is:

```c
typedef void* (*wglGetProcAddressFn)(LPCSTR);
```

Can you try changing it to:

```c
typedef PROC (*wglGetProcAddressFn)(LPCSTR);
```

**Comment by turol** [comment]
`wglGetProcAddressFn` is probably declared incorrectly. Windows requires it to have the correct calling convention, something like GLAPI or APIENTRY. Look at the GL/gl.h header (or whatever windows calls it) to see what the correct one is.

**Comment by dwilliamson** [comment]
Fixed in dfaf6ccebc000852a36f73919822509934557d82

**Comment by Megavoxel01** [comment]
Hi thanks for the fix.

I encounter another problem. When calling rmt_BeginOpenGLSample(UnscopedSample) for the second time it causes an assertion failure at line 6145. Any insight on this one?
